### PR TITLE
[INLONG-7337][Manager] Fix test Pulsar connection error

### LINKED ISSUE #7337: [Bug][Manager] Test Pulsar connection error
### What happened

Test pulsar connection error when type Muti address

image

### What you expected to happen

Should not go wrong

### How to reproduce

Test pulsar connection error when type Muti address

### Environment

_No response_

### InLong version

master

### InLong Component

InLong Manager

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by leosanqing**
### Prepare a Pull Request

- Fixes #7337 

### Motivation
Fix test pulsar connection error

### Modifications
org.apache.inlong.manager.service.cluster.InlongClusterServiceImpl#testConnection

### Verifying this change

*(Please pick either of the following options)*

- [x] This change is a trivial rework/code cleanup without any test coverage.

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java [hunk 1]
```diff
@@ -110,19 +112,34 @@ public Boolean testConnection(ClusterRequest request) {
...
+                            return false;
+                        }
+                    });
+
+            if (successConnect) {
+                return true;
+            } else {
+                throw new BusinessException("Pulsar connection failed for serviceUrl");
+            }
         } catch (Exception e) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java:143]: This catch seems unnecessary.

Comment (leosanqing) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java:143]: This is an exception of PulsarAdmin, not all of them are displayed on this change page

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java [hunk 2]
```diff
@@ -110,19 +112,34 @@ public Boolean testConnection(ClusterRequest request) {
...
+                            return true;
+                        } catch (IOException e) {
+                            String errMsg = String.format("Pulsar connection failed for AdminUrl=%s, ServiceUrl=%s",
+                                    pulsarInfo.getAdminUrl(), hostPort);
+                            LOGGER.error(errMsg, e);
+                            return false;
+                        }
+                    });
+
+            if (successConnect) {
```
Comment (fuweng11) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java:138]: Suggest to use `Preconditions.expectTrue(successConnect, "Pulsar connection failed for serviceUrl")`.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java [hunk 3]
```diff
@@ -110,19 +112,34 @@ public Boolean testConnection(ClusterRequest request) {
...
-            SocketAddress socketAddress = new InetSocketAddress(host, port);
-            Socket socket = new Socket();
-            socket.connect(socketAddress, 30000);
-            socket.close();
-            LOGGER.debug("Pulsar connection not null - connection success for AdminUrl={}, ServiceUrl={}",
-                    pulsarInfo.getAdminUrl(), pulsarInfo.getUrl());
-            return true;
+            boolean successConnect = Arrays.stream(hostPortStr.split(InlongConstants.COMMA))
+                    // If there are multiple addresses, as long as one succeeds, it's considered successful
+                    .anyMatch(hostPort -> {
```
Comment (vernedeng) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java:117]: I think it is considered successful only when all addressses are available.

**Review by vernedeng — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java (hunk 2 above)
Comment (leosanqing) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java]: done

**Comment by leosanqing** [comment]
> @leosanqing you could run `mvn spotless:apply` to fix the format error.



> @leosanqing you could run `mvn spotless:apply` to fix the format error.

done

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java (hunk 3 above)
Comment (leosanqing) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java]: done

**Review by fuweng11 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by vernedeng — APPROVED** [review]
lgtm

**Review by woofyzhao — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java [hunk 4]
```diff
@@ -100,35 +102,65 @@ public Boolean testConnection(ClusterRequest request) {
...
+                    Preconditions.expectTrue(hostPortArr.length >= 2,
+                            String.format("Pulsar ServiceUrl=%s should has ip and port, such as '127.0.0.1:6650'",
+                                    serviceUrl));
+
+                    String host = hostPortArr[0];
+                    int port = Integer.parseInt(hostPortArr[1]);
+
+                    try (Socket socket = new Socket()) {
+                        SocketAddress socketAddress = new InetSocketAddress(host, port);
+                        socket.connect(socketAddress, 30000);
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/cluster/PulsarClusterOperator.java:154]: Should we need to close the socket connection after testing?

**Review by healchow — APPROVED** [review]
LGTM

**Pull request merged**